lisa/sut_orchestrator/baremetal/bootconfig.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+from dataclasses import dataclass, field
+from pathlib import PurePosixPath
+from typing import Optional, Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+from lisa.node import quick_connect
+from lisa.tools import Cat, Sed
+from lisa.util import InitializableMixin, LisaException, field_metadata, subclasses
+from lisa.util.logger import get_logger
+
+from .schema import BootConfigSchema
+
+
+class BootConfig(subclasses.BaseClassWithRunbookMixin, InitializableMixin):
+    def __init__(
+        self,
+        runbook: BootConfigSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.boot_config_runbook: BootConfigSchema = self.runbook
+        self._log = get_logger("boot_config", self.__class__.__name__)
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return BootConfigSchema
+
+    def config(self) -> None:
+        raise NotImplementedError()
+
+
+@dataclass_json()
+@dataclass
+class PxeBootSchema(BootConfigSchema):
+    connection: Optional[schema.RemoteNode] = field(
+        default=None, metadata=field_metadata(required=True)
+    )
+    host_name: str = field(default="", metadata=field_metadata(required=True))
+    image_source: str = field(default="", metadata=field_metadata(required=True))
+    kernel_append: Optional[str] = field(default="")
+
+
+class PxeBoot(BootConfig):
+    def __init__(
+        self,
+        runbook: PxeBootSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.pxe_runbook: PxeBootSchema = self.runbook
+        self._log = get_logger("pxe_boot", self.__class__.__name__)
+        self._boot_label = "label lisa3 baremetal"
+        self._boot_dir = "/var/lib/tftpboot/"
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "pxe_boot"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return PxeBootSchema
+
+    def config(self) -> None:
+        try:
+            self._connect_to_dhcp_server()
+        except Exception as ex:
+            self._log.exception(
+                f"failed to connect {self.pxe_runbook.connection}",
+                exc_info=ex,
+            )
+
+        node_name = f"{self.pxe_runbook.host_name}"
+        host_ref = f"host {node_name}"
+        pxe_boot_config_path = self._get_pxe_config_path(node_name, host_ref)
+
+        boot_image = self.pxe_runbook.image_source.replace(
+            self._boot_dir,
+            "",
+        ).strip("/")
+        boot_entry = f"kernel {boot_image}"
```

Comment:
Does the `boot_entry` need to be some information, or just file name? If it contains different information like kernel version, it should be a separated field.

Reply:
It points to the kernel image path for bootup. Based on the requirement, it is the relative path of image

Reply:
Where is the "requirement" from? Why doesn't it support absolute paths?

Reply:
I didn't try the absolute path and just follow the existing menu format. 
Here is the boot menu for current setting

label lisa baremetal
    kernel users/paull/fitImage-gp-bm.bin

label lisav2
    kernel users/lis-test/20240819-221158-881/july-soc-gp.bin

label STOS
    kernel "users/v-sherielgin/Image_rsa.img"

label selgin-STOS-test
    kernel "users/v-sherielgin/GP_StableImages/fitImage-initramfs-ev-stable-ovl-2008-GA-gp-stos.bin"

label lisav2
    kernel users/lis-test/20240819-221158-881/july-soc-gp.bin

label STOS
    kernel "users/v-sherielgin/Image_rsa.img"

label selgin-STOS-test
    kernel "users/v-sherielgin/GP_StableImages/fitImage-initramfs-ev-stable-ovl-2008-GA-gp-stos.bin"

Reply:
Please don't share internal info here. Absolute path is more flexible and stable. It doesn't need the logic to find relative path.

Reply:
I tried to use absolutely path and it failed due to error of missing boot image file, which means the relative path is needed for the boot image file.

Reply:
ok, we have to use relative path.